From c3bc9ebd088abc80853ee25c4c17618d580a50f3 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Mon, 16 Feb 2015 17:56:27 +0100
Subject: [PATCH] leveldb: restore selinux context for xinetd conf files

---
 Makefile  |  2 +-
 leveldb.c | 41 +++++++++++++++++++++++++++++++++++++++--
 2 files changed, 40 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 76d0e64..3883d35 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@ VERSION=$(shell awk '/Version:/ { print $$2 }' chkconfig.spec)
 TAG = chkconfig-$(VERSION)
 
 CFLAGS=-g -Wall $(RPM_OPT_FLAGS) -D_GNU_SOURCE
-LDFLAGS=-g
+LDFLAGS=-g -lselinux -lsepol
 MAN=chkconfig.8 ntsysv.8 alternatives.8
 PROG=chkconfig
 BINDIR = /sbin
diff --git a/leveldb.c b/leveldb.c
index 25efb45..b33ab8e 100644
--- a/leveldb.c
+++ b/leveldb.c
@@ -27,6 +27,8 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <selinux/selinux.h>
+#include <selinux/label.h>
 
 /* Changes
    1998-09-22 - Arnaldo Carvalho de Melo <acme@conectiva.com.br>
@@ -38,6 +40,36 @@
 
 #include "leveldb.h"
 
+int selinux_restore(const char *name) {
+        struct selabel_handle *hnd = NULL;
+        struct stat buf;
+        security_context_t newcon = NULL;
+        int r = -1;
+
+        hnd = selabel_open(SELABEL_CTX_FILE, NULL, 0);
+        if (hnd == NULL)
+                goto out;
+
+        r = stat(name, &buf);
+        if (r < 0)
+                goto out;
+
+        r = selabel_lookup_raw(hnd, &newcon, name, buf.st_mode);
+        if (r < 0)
+                goto out;
+
+        r = setfilecon_raw(name, newcon);
+        if (r < 0)
+                goto out;
+
+        r = 0;
+
+ out:
+        selabel_close(hnd);
+        freecon(newcon);
+        return r;
+}
+
 int parseLevels(char * str, int emptyOk) {
     char * chptr = str;
     int rc = 0;
@@ -743,7 +775,8 @@ int setXinetdService(struct service s, int on) {
 	char tmpstr[50];
 	char *buf, *ptr, *tmp;
 	struct stat sb;
-        mode_t mode;
+    mode_t mode;
+    int r;
 	
 	if (on == -1) {
 		on = s.enabled ? 1 : 0;
@@ -790,7 +823,11 @@ int setXinetdService(struct service s, int on) {
 	}
 	close(newfd);
 	unlink(oldfname);
-	return(rename(newfname,oldfname));
+	r = rename(newfname,oldfname);
+	if (selinux_restore(oldfname) != 0)
+			fprintf(stderr, _("Unable to set selinux context for %s: %s\n"), oldfname,
+			strerror(errno));
+	return(r);
 }
 
 int doSetService(struct service s, int level, int on) {
-- 
1.8.3.1

