From 6a40c998d5e956c552febdd3bb64a6eeb235ada5 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Thu, 7 Jan 2016 11:15:06 +0100
Subject: [PATCH] leveldb: suppress error messages when selinux is turned off

---
 leveldb.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/leveldb.c b/leveldb.c
index a2c0d59..3c86ffb 100644
--- a/leveldb.c
+++ b/leveldb.c
@@ -69,6 +69,15 @@ int selinux_restore(const char *name) {
                 selabel_close(hnd);
         if (newcon)
                 freecon(newcon);
+
+        /* Lets ignore any errors when selinux is disabled.
+         * We still want to run the previous code though,
+         * since we only need selinux policy.
+         * Selinux itself can be turned off.
+         */
+        if (!is_selinux_enabled())
+                 return 0;
+
         return r;
 }
 
-- 
2.4.3

